app/Providers/RouteServiceProvider.php

```diff
@@ -39,8 +40,11 @@ public function boot()
     {
         parent::boot();
 
-        Route::bind('contact', function ($value) {
+        $ID_hasher = new ID_hasher();
+
+        Route::bind('contact', function ($value) use ($ID_hasher){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "contact" 11 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1192)